src/DssSpell.sol

```diff
@@ -47,17 +87,118 @@ contract DssSpellAction is DssAction, DssSpellCollateralOnboardingAction {
     function actions() public override {
         // onboardNewCollaterals();
 
-        // Update TUSD implementation
-        // https://forum.makerdao.com/t/2022-03-25-adding-support-for-the-new-tusd-implementation-address/14189
-        address MCD_JOIN_TUSD_A = DssExecLib.getChangelogAddress("MCD_JOIN_TUSD_A");
-        GemJoin6Like(MCD_JOIN_TUSD_A).setImplementation(TUSD_PREV_IMPL, 0);
-        GemJoin6Like(MCD_JOIN_TUSD_A).setImplementation(TUSD_NEXT_IMPL, 1);
-
-        // Authorize ESM on DssFlash
-        // https://forum.makerdao.com/t/proposed-security-fix-to-be-added-in-the-march-25th-2022-executive-spell
-        address MCD_FLASH = DssExecLib.getChangelogAddress("MCD_FLASH");
-        address MCD_ESM = DssExecLib.esm();
-        DssExecLib.authorize(MCD_FLASH, MCD_ESM);
+        // Core Unit DAI Budget Transfers
+        // GOV-001 - 30,000 DAI - 0x01D26f8c5cC009868A4BF66E268c17B057fF7A73 https://forum.makerdao.com/t/mip40c3-sp59-govalpha-budget-2022-23/13144
+        DssExecLib.sendPaymentFromSurplusBuffer(GOV_WALLET_1,  30_000);
+        // IS-001 - 348,452.30 DAI - 0xd1F2eEf8576736C1EbA36920B957cd2aF07280F4 https://github.com/makerdao/mips/pull/463/files
+        // Rounded up from 348,452.30 to 348,453
+        DssExecLib.sendPaymentFromSurplusBuffer(ISCU_WALLET,  348_453);
+        // RWF-001 - 2,055,000 DAI - 0x96d7b01Cc25B141520C717fa369844d34FF116ec https://mips.makerdao.com/mips/details/MIP40c3SP61#transactions
+        DssExecLib.sendPaymentFromSurplusBuffer(RWF_WALLET,  2_055_000);
+
+        // VEST.restrict( Only recipient can request funds
+        //     VEST.create(
+        //         Recipient of vest,
+        //         Total token amount of vest over period,
+        //         Start timestamp of vest,
+        //         Duration of the vesting period (in seconds),
+        //         Length of cliff period (in seconds),
+        //         Manager address
+        //     )
+        // );
+
+        // Core Unit DAI Budget Streams
+        // GOV-001 | 2022-04-01 to 2023-04-01 | 1,079,793 DAI | 0x01D26f8c5cC009868A4BF66E268c17B057fF7A73 https://forum.makerdao.com/t/mip40c3-sp59-govalpha-budget-2022-23/13144
+        MCD_VEST_DAI.restrict(
+            MCD_VEST_DAI.create(
+                GOV_WALLET_1,
+                1_079_793 * WAD,
+                APR_01_2022,
+                ONE_YEAR,
+                0,
+                GOV_WALLET_1
```

Comment:
Not sure we want to give the wallet self-yank ability.  Have we discussed this?  It will activate the `yank()` codepath for the vesting wallet, and I personally have not thought through all the implications there.
```suggestion
                address(0)
```

Reply:
That is, if we treat the wallet as adversarial, can they gain any advantage with this `yank()` ability?  We haven't allowed this in the past IIRC.

Reply:
i brought this up in the watercooler yesterday and no one seemed to have an issue

also raised with GovAlpha in #new-spells: https://discord.com/channels/893112320329396265/897483518316265553/959104957007671307

Reply:
I think it was probably an oversight that we don't allow a recipient to yank their own vest.

This was discussed with govalpha and they've ok'ed it.

I realized that there is an adversarial condition though, if a recipient moves their stream to rotate keys or because they suspect that their keys were compromised this does give the ability for the attacker to yank the stream, even after it's been moved.

When we update for the end issue we should probably add the ability for the recipient to yank their own vest such that the ability moves when the stream moves.

Reply:
sounds good. i'll revert to `address(0)` for now

Reply:
If a core unit's keys are compromised I'm not sure that yanking the stream and starting over is the worst option though.

Reply:
lets continue the discussion outside the perspective of this spell. adding it to this spell doesn't solve the problem for all streams if we do want them to be self-managed, no need to rush into a spell imo